src/LinAlg/hiopLinAlgFactory.cpp

```diff
@@ -93,6 +98,14 @@ hiopVector* LinearAlgebraFactory::create_vector(const std::string& mem_space,
   const std::string mem_space_upper = toupper(mem_space);
   if(mem_space_upper == "DEFAULT") {
     return new hiopVectorPar(glob_n, col_part, comm);
+  } else if(mem_space_upper == "CUDA") {
+    #ifdef HIOP_USE_CUDA
+      return new hiopVectorCuda(glob_n, col_part, comm);
+    #else
+      assert(false && "requested memory space not available because Hiop was not"
+                      "built with CUDA support");
+      return new hiopVectorPar(glob_n, col_part, comm);
```

Comment:
Similarly below, if you assert false, doesn't HiOp exit? Perhaps I am just misremembering how assert works, but if this is intended to warn gracefully, maybe we should reconsider.

Reply:
you are correct. The return function doesn't do anything here. I follow Asher's way to implement this part of code (see other lines in this file). I guess he added return function here to avoid compilation warning.

Reply:
I think throwing exceptions and not catching them would be equivalent to assertions used in HiOp, but would be cleaner and more flexible solution going forward.

Reply:
created issue #577